### DISCUSSION TITLE
Can we know the original type of `SearchParamDefinition`?

### DISCUSSION THREAD

**Discussion by whyfate** [FirelyTeam/firely-net-sdk#2010] [General]
The original data type of `Token` may be `Identifier`, `Coding`, `Code`, etc. http://hl7.org/fhir/search.html#token.
It would be great if we could get the original data type in `SearchParamDefinition`.

**Comment by ewoutkramer** [comment]
That's pretty hard, since there's no 1-on-1 relationship between the search parameter (which is defined as a FhirPath expression) and the elements used by this expression. Sure, if the FhirPath expression is a simple walk into the instance to a single element, you could lookup the datatype of the element in the StructureDefinition, but -although common- this can only be done for a subset of the search parameters. And it is still not trivial.

**Comment by whyfate** [comment]
Thanks, but this is really bad news.
